### ISSUE TITLE
Support for OpenAPI complex parameter using content to define serialization:

### ISSUE THREAD

**Issue by zam6ak** [seriousme/fastify-openapi-glue#603]
In OpenAPI spec [1] a parameter can be defined using `content` to define serialization
Example from spec:

```
in: query
name: coordinates
content:
  application/json:
    schema:
      type: object
      required:
        - lat
        - long
      properties:
        lat:
          type: number
        long:
          type: number
```

When I try this, I get 

```
X:\code\projects\api-fastify-poc\node_modules\fastify\lib\route.js:416
              throw new FST_ERR_SCH_VALIDATION_BUILD(opts.method, url, error.message)
                    ^
FastifyError [Error]: Failed building the validation schema for GET: /v1/people, due to error schema is invalid: data/properties/filterstrong must be object,boolean
    at Boot.<anonymous> (X:\code\projects\api-fastify-poc\node_modules\fastify\lib\route.js:416:21)
    at Object.onceWrapper (node:events:632:28)
    at Boot.emit (node:events:530:35)
    at X:\code\projects\api-fastify-poc\node_modules\avvio\boot.js:102:12
    at X:\code\projects\api-fastify-poc\node_modules\avvio\boot.js:437:7
    at done (X:\code\projects\api-fastify-poc\node_modules\avvio\lib\plugin.js:226:5)
    at check (X:\code\projects\api-fastify-poc\node_modules\avvio\lib\plugin.js:250:9)
    at node:internal/process/task_queues:140:7
    at AsyncResource.runInAsyncScope (node:async_hooks:206:9)
    at AsyncResource.runMicrotask (node:internal/process/task_queues:137:8) {
  code: 'FST_ERR_SCH_VALIDATION_BUILD',
  statusCode: 500
```

I ran the spec through the validator that this project uses and it returns as valid

```
npx -p @seriousme/openapi-schema-validator validate-api spec/oapi-v1.yml
{
        "valid": true
}
```

[1] https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#parameter-object

**Comment by seriousme** [comment]
Hi,

thanks for asking!
Currently the parser does not support the `content` fieldname for `in: query"  while the spec says that it should be possible (as confirmed by the schema validator).
If you have control over the openapi schema then the short term fix is to use `schema` instead of `content`.

I will look into it.

Kind regards,
Hans

**Comment by seriousme** [comment]
Released as 4.7.1 on NPM.
Can you please verify if this solves your problem?

Kind regards,
Hans
